.github/workflows/lint-pr-titles.yaml

```diff
@@ -0,0 +1,25 @@
+name: PR Title Lint
+on:
+  pull_request:
+    types: [opened, edited, reopened, synchronize]
+permissions:
+  pull-requests: read
+jobs:
+  pr-title-lint:
+    runs-on: ubuntu-latest
+    name: Lint PR Title
+    steps:
+      - name: Check PR Title
+        env: 
+          TITLE: ${{ github.event.pull_request.title}}
+        run: |
+            if [[ "${TITLE,,}" =~ ^\[?\(?wip\]?\)? ]]; then
+            echo "PR is marked as a WIP. Skipping validation."
+            exit 0
+            elif (echo "${TITLE,,}" | grep -P "^(fix|feat|build|chore|docs|test|refactor|ci|localize|bump|revert)(\(([A-Za-z0-9])+\))?:"); then
```

Comment:
tbh my scripting knowledge is at a beginner level so there's probably a better way to handle this...but idea is there...